cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/GetContainerIdOfSegmentCommand.java

```diff
@@ -0,0 +1,85 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package io.pravega.cli.admin.segmentstore;
+
+
+import io.pravega.cli.admin.CommandArgs;
+import io.pravega.client.connection.impl.ConnectionPool;
+import io.pravega.controller.server.SegmentHelper;
+import io.pravega.shared.NameUtils;
+import io.pravega.shared.protocol.netty.WireCommands;
+import io.pravega.shared.segment.SegmentToContainerMapper;
+import com.google.common.base.Preconditions;
+import lombok.Cleanup;
+import org.apache.curator.framework.CuratorFramework;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Gets the container id of the given segment.
+ *  The arguments for the command that is the fully qualified segment name.
+ */
+public class GetContainerIdOfSegmentCommand extends SegmentStoreCommand {
+
+    private final SegmentToContainerMapper segToConMapper;
+
+    public GetContainerIdOfSegmentCommand(CommandArgs args) {
+        super(args);
+        int containerCount = getServiceConfig().getContainerCount();
+        this.segToConMapper = new SegmentToContainerMapper(containerCount, true);
+    }
+
+    @Override
+    public void execute() throws Exception {
+        ensureArgCount(1);
+        @Cleanup
+        CuratorFramework zkClient = createZKClient();
+        @Cleanup
+        ConnectionPool pool = createConnectionPool();
+        @Cleanup
+        SegmentHelper segmentHelper = instantiateSegmentHelper(zkClient, pool);
+        final String fullyQualifiedSegmentName = getArg(0);
+
+        String[] inputParam = fullyQualifiedSegmentName.split("/");
+        Preconditions.checkArgument(inputParam.length == 3, "Invalid qualified-segment-name  '%s'", fullyQualifiedSegmentName);
+        String scope = inputParam[0];
+        String stream = inputParam[1];
+        String[] segmentArr = inputParam[2].split("\\.");
+
+        try {
+            int segmentNumber = Integer.parseInt(segmentArr[0]);
+            int epoch = Integer.parseInt(segmentArr[2]);
+            long segmentId = NameUtils.computeSegmentId(segmentNumber, epoch);
+            CompletableFuture<WireCommands.StreamSegmentInfo> segmentInfoFuture = segmentHelper.getSegmentInfo(scope, stream, segmentId, super.authHelper.retrieveMasterToken(), 0L);
+            segmentInfoFuture.join();
+            
+            int containerId = segToConMapper.getContainerId(fullyQualifiedSegmentName);
+            output("Container Id for the given Segment is :" + containerId);
+        } catch (Exception ex) {
+
+            output("Error : %s Segment does not exist %s", fullyQualifiedSegmentName, ex);
```

Comment:
If exception occurs here, does it always mean Segment does not exist ?

Reply:
Yes Ankur, as the other conditions for fully qualified segment name and the number of arguments is already checked before. This was added to check if the segment exists or not.

Reply:
What I meant is, may be there could be connection failure exception in which case this message looks wrong.

Reply:
Yes, I have updated my PR with the change now.